packages/purgecss/src/index.ts

```diff
@@ -214,17 +214,28 @@ function isTagFound(
 }
 
 /**
- * Returns true if the selector is inside a pseudo class
+ * Returns true if the node is a pseudo class
  * (e.g. :nth-child, :nth-of-type, :only-child, :not)
- * @param selector selector
+ * @param node node
  */
-function isInPseudoClass(selector: selectorParser.Node): boolean {
-  return (
-    (selector.parent &&
-      selector.parent.type === "pseudo" &&
-      selector.parent.value.startsWith(":")) ||
-    false
-  );
+function isPseudoSelector(node: selectorParser.Node): boolean {
+  return node.type === "pseudo" && node.value.startsWith(":");
+}
+
+/**
+ * Returns true if the node is inside a pseudo class
+ * (e.g. :nth-child, :nth-of-type, :only-child, :not)
+ * @param node selector
+ */
+function isInPseudoClass(node: selectorParser.Node): boolean {
+  if (isPseudoSelector(node.parent as selectorParser.Node)) {
+    return true;
+  }
+  const selector = node as selectorParser.Selector;
+  if (selector.nodes.length === 2 && isPseudoSelector(selector.nodes[0])) {
```

Comment:
This condition seems to be specific to the example but will not work consistently. Is there a specific reason why you are checking if it has 2 selectors?
It would mean it will not work for `<p class="bar hello"` `:not(.foo-bar).bar.hello`?

Reply:
Thanks for noticing this moment!

Is `selector.nodes.length > 1` the right condition for handling all cases?

Reply:
@Ffloriel do we need to add multiple selectors case into the test?